Doctrine/Phpcr/AutoRouteListener.php

```diff
@@ -73,6 +74,13 @@ public function onFlush(ManagerEventArgs $args)
                     $dm->persist($autoRoute);
                     $uow->computeChangeSets();
                 }
+
+                // reset locale to the original locale
+                if (null !== $locale) {
```

Comment:
you could compare locale with $uow->getCurrentLocale to check if its needed to reset the translation